src/main/java/core/basesyntax/IsoscelesTrapezoid.java

```diff
@@ -0,0 +1,30 @@
+package core.basesyntax;
+
+import java.text.DecimalFormat;
+
+public class IsoscelesTrapezoid extends Figure {
+    private int baseOne;
+    private int baseTwo;
+    private int height;
+    private String color;
+
+    public IsoscelesTrapezoid(Figure.FigureSupplier figureSupplier, ColorSupplier colorSupplier) {
+        this.baseOne = figureSupplier.getRandomFigure();
+        this.baseTwo = figureSupplier.getRandomFigure();
+        this.height = figureSupplier.getRandomFigure();
+        this.color = colorSupplier.getRandomColor();
+    }
```

Comment:
pass arguments to the constructor to init your fields, don't use anything from figure supplier